projects/packages/boost-core/src/lib/class-utils.php

```diff
@@ -133,11 +133,24 @@ public static function send_wpcom_request( $method, $endpoint, $args = null, $bo
 				$code
 			);
 
+			/*
+			 * WordPress.com passes statusCode and error back from Boost Cloud.
+			 * WordPress.com ISA returns code and message.
+			 */
 			// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-			$err_code = empty( $data['statusCode'] ) ? 'http_error' : $data['statusCode'];
-			$message  = empty( $data['error'] ) ? $default_message : $data['error'];
-
-			return new \WP_Error( $err_code, $message );
+			if ( isset( $data['statusCode'] ) && isset( $data['error'] ) ) {
+				// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$data_code    = $data['statusCode'];
+				$data_message = $data['error'];
+			} elseif ( isset( $data['code'] ) && isset( $data['message'] ) ) {
+				$data_code    = $data['code'];
+				$data_message = $data['message'];
+			}
+
+			$error_code = empty( $data_code ) ? 'http_error' : $data_code;
+			$message    = empty( $data_message ) ? $default_message : $data_message;
+
+			return new \WP_Error( $error_code, $message );
```

Comment:
I wonder if we should make it consistent on the wpcom part instead.

Reply:
Maybe. Though I think WPCOM has it right. The cloud has it wrong 😅

Reply:
Oh wait. But if we change wpcom, does that mean that wpcom will return statusCode and error?

I think this might cause more harm than good...

Reply:
It turns out that the change to boost core was necessary to figure out that the response is `not-found` and not a generic error.

Reply:
Left a comment detailing why we need the band-aid I removed in afa7179.

p1734690571772729-slack-C016BBAFHHS

Reply:
I've added the band-aid back into Boost.

Reply:
Updated (d22ffed) comment explaining error normalization.